src/import/kv_importer.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs;
+use std::fmt;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex};
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use uuid::Uuid;
+use kvproto::importpb::*;
+
+use config::DbConfig;
+use util::collections::HashMap;
+
+use super::engine::*;
+use super::{Config, Error, Result};
+
+pub type Token = usize;
+
+struct Inner {
+    engines: HashMap<Uuid, Arc<EngineFile>>,
+    clients: HashMap<Token, Arc<EngineFile>>,
+}
+
+pub struct KVImporter {
+    dir: EngineDir,
+    token: AtomicUsize,
+    inner: Mutex<Inner>,
+}
+
+impl KVImporter {
+    pub fn new(cfg: Config, opts: DbConfig) -> Result<KVImporter> {
+        let dir = EngineDir::new(&cfg.import_dir, opts)?;
+        Ok(KVImporter {
+            dir: dir,
+            token: AtomicUsize::new(1),
+            inner: Mutex::new(Inner {
+                engines: HashMap::default(),
+                clients: HashMap::default(),
+            }),
+        })
+    }
+
+    pub fn token(&self) -> Token {
+        self.token.fetch_add(1, Ordering::SeqCst)
+    }
+
+    fn insert(&self, token: Token, engine: Arc<EngineFile>) {
```

Comment:
The function name is misleading. It can be easily mis-interpreted as something related to the KvEngine's data insert.

Reply:
Well, `HashMap` has an insert method too. `KVImporter` is a struct to maintain the relationship between clients and engines, it has nothing to do with data insert/remove.

Reply:
For a `HashMap`'s insert, we know exactly what it inserts. When it is interpreted as "insert into HashMap" the purpose is obvious. However, for `KVImporter`'s insert, usually we have no idea unless take a look at the implementation. I mean, "insert into KVImporter" is confusing, because we neither know what KVImporter really does, nor what insert really does, according to their names. I suggest to either document as comment for at least one of them, or change the name of the function so that it is self-explaining.